### DISCUSSION TITLE
Missing values for Drive table?

### DISCUSSION THREAD

**Discussion by jun3280net** [teslamate-org/teslamate#1719] [Q&A]
Hi there, was wondering if anybody was experiencing this issue related to their Drives?  My Drive's starting from Home (Geofenced) are triggering a Drive record at the proper time, but the rest of the fields are not populated.  10 minutes later, another Drive record is created with all values populated as expected.  

Other threads I've read noted drives missing 10 minutes and starting at the wrong location - which I believe is a different issue (eg Drive record triggered 10 minutes late).  In my case, the triggers are at the correct time.  It happens once a day - I suspect when the vehicle wakes up from SLEEP.  

While, I think most the of values can be captured from the records before and after.  This looks almost like it might be a bug, as the **end_date** is not being captured for the record in question.

**Drive Table**
![image](https://user-images.githubusercontent.com/30555695/121815783-5c5f1400-cc35-11eb-8ff8-ef8edf5e79ee.png)

**Timeline Table**
![image](https://user-images.githubusercontent.com/30555695/121816047-a1d01100-cc36-11eb-9e66-06e09016da7c.png)

Any thoughts on this would be greatly appreciated.

**Comment by init6-git** [comment]
I've got something similar. Trips seem to be split into multiple drives even though I have only been on one drive.
<img width="1139" alt="Screenshot 2021-06-15 at 11 55 12" src="https://user-images.githubusercontent.com/85936842/122041318-b9db9800-cdd0-11eb-8701-f9efb8501298.png">
When I look on drives the blank drive does not appear - only the part drive. 
I have only seen this behaviour since last update.

**Comment by adriankumpf** [comment]
@jun3280net when such a drive is recorded and you suspect a bug, take a look at the TeslaMate logs. It's hard to tell what's going on otherwise.

**Reply by jun3280net** [reply]
Hi @adriankumpf, below is the log correlated with June 12th (ID 2393) captured in the Drive Table I screenshot with my original post.

![image](https://user-images.githubusercontent.com/30555695/122164340-e7a5f880-ce33-11eb-95b9-5eca0dbdd421.png)

On that note, I did a "docker-compose pull" shortly after I posted, and its been good for the past 3 days.  That said, I had updated probably just a few days before, which might have contributed to this - as it was working fine before that.

**Reply by adriankumpf** [reply]
The errors shown in the screenshot are returned by Tesla's API As a result, TeslaMate could not query any data.

**Comment by sunflower-green** [comment]
I'm experiencing the same issue it seems. I'm running the latest version 1.23.3.
The issue started recently with the release of the new version. Teslamate has been running flawlessly since january.

Some drives are completely gone/not visible from the "Drives" dashboard. Though, when looking at the "Trip" dashboard I can see that Teslamate indeed did record the drive (route on map visible). 

![image](https://user-images.githubusercontent.com/86048675/122367638-e3c1c580-cf5c-11eb-952d-19d1be8a41e4.png)

The drives that are missing (from Drives) seems to be missing the Start and Destination address.
The time is correct on those drives. I can click on the time-field and view the drive with included data.

See attached log during that time-span

> teslamate_1  | 2021-06-16 17:49:47.979 [warn] Too many disconnects from streaming API
teslamate_1  | 2021-06-16 17:49:47.980 car_id=1 [info] Creating new connection …
teslamate_1  | 2021-06-16 17:49:47.980 car_id=1 [info] Disconnecting ...
teslamate_1  | 2021-06-16 17:49:48.008 car_id=1 [info] Connecting ...
teslamate_1  | 2021-06-16 17:51:09.439 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/XXXXX/vehicle_data -> 408 (9283.522 ms)
teslamate_1  | 2021-06-16 17:51:10.070 car_id=1 [warn] Discarded incomplete fetch result
teslamate_1  | 2021-06-16 18:03:41.925 [error] ** (RuntimeError) Client Error: %{"error_type" => "client_error", "msg_type" => "data:error", "tag" => "162478183845", "value" => "owner_api error: `Got HTTP status: 503 Service Unavailable for . Response: upstream connect error or disconnect/reset before headers. reset reason: connection failure`"}
teslamate_1  |     (teslamate 1.23.3) lib/tesla_api/stream.ex:150: TeslaApi.Stream.handle_frame/2
teslamate_1  |     (websockex 0.4.3) lib/websockex.ex:1113: WebSockex.try_callback/3
teslamate_1  |     (websockex 0.4.3) lib/websockex.ex:797: WebSockex.common_handle/4
teslamate_1  |     (stdlib 3.15.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1  |
teslamate_1  | 2021-06-16 18:03:50.380 car_id=1 [info] Start / :online
teslamate_1  | 2021-06-16 18:03:50.493 car_id=1 [info] Connecting ...
teslamate_1  | 2021-06-16 18:03:50.497 car_id=1 [info] Driving / Start
teslamate_1  | 2021-06-16 18:11:01.104 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (177.399 ms)
teslamate_1  | 2021-06-16 18:11:01.597 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (479.004 ms)
teslamate_1  | 2021-06-16 18:11:01.647 car_id=1 [info] Driving / Ended / 6 km – 7 min
teslamate_1  | 2021-06-16 18:11:01.647 car_id=1 [info] Start / :online
teslamate_1  | 2021-06-16 18:11:05.556 car_id=1 [info] Received stream data: :inactive
teslamate_1  | 2021-06-16 18:14:42.520 car_id=1 [info] Suspending logging
teslamate_1  | 2021-06-16 18:17:27.767 car_id=1 [info] Fetching vehicle state ...
teslamate_1  | 2021-06-16 18:21:51.498 car_id=1 [info] Fetching vehicle state ...
teslamate_1  | 2021-06-16 18:29:06.994 car_id=1 [info] Fetching vehicle state ...
teslamate_1  | 2021-06-16 18:32:21.349 car_id=1 [info] Driving / Start
teslamate_1  | 2021-06-16 18:48:41.902 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/XXXXX/vehicle_data -> 408 (9560.812 ms)
teslamate_1  | 2021-06-16 18:48:42.243 car_id=1 [warn] Discarded incomplete fetch result
teslamate_1  | 2021-06-16 18:57:37.184 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/XXXXX/vehicle_data -> 408 (11205.581 ms)
teslamate_1  | 2021-06-16 18:57:38.110 car_id=1 [warn] Discarded incomplete fetch result
teslamate_1  | 2021-06-16 18:57:57.620 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/XXXXX/vehicle_data -> 408 (9507.376 ms)
teslamate_1  | 2021-06-16 18:57:57.837 car_id=1 [warn] Discarded incomplete fetch result
teslamate_1  | 2021-06-16 18:58:36.830 car_id=1 [info] Received stream data: :inactive
teslamate_1  | 2021-06-16 19:08:55.404 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (179.343 ms)
teslamate_1  | 2021-06-16 19:08:55.942 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (530.966 ms)
teslamate_1  | 2021-06-16 19:08:56.009 car_id=1 [info] Driving / Ended / 29 km – 37 min
teslamate_1  | 2021-06-16 19:08:56.009 car_id=1 [info] Start / :online
teslamate_1  | 2021-06-16 19:09:58.885 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (218.336 ms)
teslamate_1  | 2021-06-16 19:09:58.948 car_id=1 [info] Charging / SOC: 35% / Hjem
teslamate_1  | 2021-06-16 19:09:58.949 car_id=1 [info] Disconnecting ...
teslamate_1  | 2021-06-16 19:12:15.518 [warn] GET https://owner-api.teslamotors.com/api/1/vehicles/XXXXX/vehicle_data -> 503 (427.154 ms)
teslamate_1  | 2021-06-16 19:12:15.520 [warn] TeslaApi.Error / "upstream connect error or disconnect/reset before headers. reset reason: connection failure"
teslamate_1  | 2021-06-16 19:12:15.521 car_id=1 [error] Error / :unknown
teslamate_1  | 2021-06-16 19:22:45.295 [warn] GET https://owner-api.teslamotors.com/api/1/vehicles/XXXXX/vehicle_data -> 503 (428.974 ms)
teslamate_1  | 2021-06-16 19:22:45.296 [warn] TeslaApi.Error / "upstream connect error or disconnect/reset before headers. reset reason: connection failure"
teslamate_1  | 2021-06-16 19:22:45.297 car_id=1 [error] Error / :unknown
teslamate_1  | 2021-06-16 19:35:40.603 [info] GET /
teslamate_1  | 2021-06-16 19:35:40.633 [info] Sent 200 in 29ms
teslamate_1  | 2021-06-16 20:08:42.491 [info] GET /

> 

 @adriankumpf
This discussion should maybe belong to "Issues"?

**Reply by jun3280net** [reply]
@menkromsen I was definitely on the latest version like you.  However, since doing the "docker-compose pull" and reboot I have not experienced this issue with DRIVES for the past 3-4 days.  But I agree - there might might have been a bug somewhere , as Teslamate was running great for me as well prior to that update.  That said - my issues related to the first 10 minutes of a DRIVE, however you seem to have multiple consecutive blank DRIVES.

**Reply by sunflower-green** [reply]
@jun3280net 
Yes, our issues are not completely similar -but something funky is going on with the latest Teslamate release...
It seems that issues #1533 and #1731 are related.

My last drives are recorded as normal but they are rather short. My last long trips have gone bad.
I can't see why "docker-compose pull" should fix this issue as that command should only pull the latest published Teslamate-container from Docker as far as I know? Maybe a reboot is all that is needed, but I'm not too sure about that either :P
Hopefully Adrian may  find a solution to why this is happening.

**Reply by enigma2k** [reply]
@menkromsen I'm experiencing the same problem as you

**Comment by eliluong** [comment]
I had the same problem today for the first time; missed entire 20-ish minute drive except for the last minute. Mostly API errors, I think. I just now updated from 1.22.3 to 1.22.4, we'll see what happens.

[teslamate-log-incomplete.txt](https://github.com/adriankumpf/teslamate/files/6679676/teslamate-log-incomplete.txt)
